packages/astro/src/content/runtime.ts

```diff
@@ -64,11 +68,19 @@ export function createGetCollection({
 		} else if (collection in dataCollectionToEntryMap) {
 			type = 'data';
 		} else if (store.hasCollection(collection)) {
-			return [...store.values<DataEntry>(collection)].map((entry) => ({
-				...entry,
-				collection,
-				render: () => renderEntry(entry),
-			}));
+			// @ts-expect-error	virtual module
+			const { default: imageAssetMap } = await import('astro:asset-imports');
+
+			return [...store.values<DataEntry>(collection)].map((entry) => {
+				if (entry.filePath) {
+					entry.data = updateImageReferencesInData(entry.data, entry.filePath, imageAssetMap);
```

Comment:
Isn't this mutating the original entry?

Reply:
No, it uses `Traverse.map`, which returns a new object. I could use the method that mutates it in place, but that gives me an instinctive ick. It might probably save memory though, but it's probably marginal.

Reply:
Ah, got it. I'll change that.